### PR TITLE
Error out if the client sends a not-yet-used version number in 'newer'.

### PR DESCRIPTION
**Pull request by rfk**
Per Issue 28.  Not a full solution, but maybe a useful first step.

### CODE REVIEW

File: routes/syncstore.js [hunk 1]
```diff
@@ -150,6 +150,12 @@ function getItems(request) {
       }
     }
 
+    // If they sent a 'newer' param that's ahead of the current state
+    // of the server, they're likely doing something wrong.  Error out.
+    if (request.query.newer && request.query.newer > res.version) {
+      return request.replay(Hapi.Error.badRequest('unseen version number'));
```
Comment (zaach) [inline routes/syncstore.js]: Looks like that should say `reply` rather than `replay`.

File: test/integration/syncstore.js [hunk 2]
```diff
@@ -173,5 +173,16 @@ describe('syncstore web api', function() {
     });
   });
 
+  it('errors if client sends a not-yet-used version number', function(done) {
+    makeRequest('GET', '/storage/col1', function(res) {
+      assert.equal(res.statusCode, 200);
+      var newer = res.version + 1;
+      makeRequest('GET', '/storage/col1?newer='+newer, function(res) {
```
Comment (zaach) [inline test/integration/syncstore.js:180]: `newer` is returning NaN here, so the test still passes because NaN causes a validation error.

**Comment by rfk** [comment]
Whoops, good catch @zaach.  Fixed the typo and the test.

**Comment by zaach** [comment]
r+

**Pull request merged**